pipeline-runner/R/gem2s-7-upload_to_aws.R

```diff
@@ -1,97 +1,93 @@
+                   object = charToRaw(cell_sets_data),
+                   key = experiment_id
   )
 
-  color_pool <- get_color_pool()
-  samples_set <- samples_sets(input, output_dir, color_pool)
-
-  # remove used colors from pool
-  used <- seq_along(samples_set$children)
-  color_pool <- color_pool[-used]
-
-  # Design cell_set meta_data for DynamoDB
-  cell_sets <- c(list(scratchpad), list(samples_set))
-
-  if ("metadata" %in% names(config)) {
-    cell_sets <- c(cell_sets, meta_sets(output_dir, color_pool))
-  }
+  # seurat object to s3
+  message("Uploading Seurat Object to S3 ...")
+  fpath <- file.path(tempdir(), 'experiment.rds')
+  saveRDS(scdata, fpath, compress = FALSE)
 
-  cell_sets <- list(cellSets = cell_sets)
+  put_object_in_s3_multipart(pipeline_config,
+                             bucket = pipeline_config$source_bucket,
+                             object = fpath,
+                             key = file.path(experiment_id, "r.rds")
+  )
 
-  print(paste("Experiment name is", config$name))
+  cluster_env <- pipeline_config$cluster_env
 
   experiment_data <- list(
     apiVersion = "2.0.0-data-ingest-seurat-rds-automated",
     experimentId = experiment_id,
     experimentName = config$name,
     meta = list(
       organism = config$organism,
-      type = config$input[["type"]]
+      type = config$input$type
     ),
     processingConfig = config_dataProcessing
   )
 
-  # cell sets file to s3
-  cell_sets_data <- RJSONIO::toJSON(cell_sets)
+  res <- list(
+    data = list(
+      item = experiment_data,
+      table = pipeline_config$experiments_table),
+    output = list())
 
-  put_object_in_s3(pipeline_config,
-    bucket = pipeline_config$cell_sets_bucket,
-    object = charToRaw(cell_sets_data),
-    key = experiment_id
-  )
+  return(res)
+}
 
-  # seurat object to s3
-  put_object_in_s3_multipart(pipeline_config,
-    bucket = pipeline_config$source_bucket,
-    object = file.path(output_dir, "experiment.rds"),
-    key = file.path(experiment_id, "r.rds")
+# creates initial cell sets object
+get_cell_sets <- function(scdata, input) {
+
+  scratchpad <- list(
+    key = "scratchpad",
+    name = "Custom cell sets",
+    rootNode = TRUE,
+    children = list(),
+    type = "cellSets"
   )
 
-  cluster_env <- pipeline_config$cluster_env
-  print(sprintf("Experiment ID: %s uploaded to %s.", experiment_id, cluster_env))
+  color_pool <- get_color_pool()
+  samples_set <- samples_sets(input, scdata, color_pool)
 
-  data <- list(
-    item = experiment_data,
-    table = pipeline_config$experiments_table
-  )
+  # remove used colors from pool
+  used <- seq_along(samples_set$children)
+  color_pool <- color_pool[-used]
 
-  if (cluster_env == "production") {
-    print(sprintf("https://scp.biomage.net/experiments/%s/data-exploration", experiment_id))
+  # Design cell_set meta_data for DynamoDB
+  cell_sets <- c(list(scratchpad), list(samples_set))
+
+  # TODO: refactor meta_sets
```

Comment:
take a look if this is not something that's already done

Reply:
Hadn't done yet - nice catch!